compactor/src/parquet_file_combining.rs

```diff
@@ -0,0 +1,579 @@
+    use arrow_util::assert_batches_sorted_eq;
+    use data_types::{ColumnType, PartitionKey, PartitionParam};
+    use iox_tests::util::{TestCatalog, TestParquetFileBuilder, TestTable};
+    use parquet_file::ParquetFilePath;
+    use test_helpers::assert_error;
+
+    struct TestSetup {
+        catalog: Arc<TestCatalog>,
+        table: Arc<TestTable>,
+        candidate_partition: PartitionCompactionCandidateWithInfo,
+        parquet_files: Vec<ParquetFile>,
+    }
+
+    async fn test_setup() -> TestSetup {
+        let catalog = TestCatalog::new();
+        let ns = catalog.create_namespace("ns").await;
+        let sequencer = ns.create_sequencer(1).await;
+        let table = ns.create_table("table").await;
+        table.create_column("field_int", ColumnType::I64).await;
+        table.create_column("tag1", ColumnType::Tag).await;
+        table.create_column("tag2", ColumnType::Tag).await;
+        table.create_column("tag3", ColumnType::Tag).await;
+        table.create_column("time", ColumnType::Time).await;
+        let table_schema = table.catalog_schema().await;
+
+        let partition = table
+            .with_sequencer(&sequencer)
+            .create_partition("2022-07-13")
+            .await;
+
+        // The sort key comes from the catalog and should be the union of all tags the
+        // ingester has seen
+        let sort_key = SortKey::from_columns(["tag1", "tag2", "tag3", "time"]);
+        let partition = partition.update_sort_key(sort_key.clone()).await;
+
+        let candidate_partition = PartitionCompactionCandidateWithInfo {
+            table: Arc::new(table.table.clone()),
+            table_schema: Arc::new(table_schema),
+            namespace: Arc::new(ns.namespace.clone()),
+            candidate: PartitionParam {
+                partition_id: partition.partition.id,
+                sequencer_id: partition.partition.sequencer_id,
+                namespace_id: ns.namespace.id,
+                table_id: partition.partition.table_id,
+            },
+            sort_key: Some(sort_key),                   //NGA TODO
+            partition_key: PartitionKey::from("to-do"), // NGA TODO
+        };
+
+        let lp = vec![
+            "table,tag2=PA,tag3=15 field_int=1601i 30000",
+            "table,tag2=OH,tag3=21 field_int=21i 36000",
+        ]
+        .join("\n");
+        let builder = TestParquetFileBuilder::default()
+            .with_line_protocol(&lp)
+            .with_compaction_level(CompactionLevel::FileNonOverlapped); // Prev compaction
+        let level_1_file = partition.create_parquet_file(builder).await;
+
+        let lp = vec![
+            "table,tag1=WA field_int=1000i 8000", // will be eliminated due to duplicate
+            "table,tag1=VT field_int=10i 10000",
+            "table,tag1=UT field_int=70i 20000",
+        ]
+        .join("\n");
+        let builder = TestParquetFileBuilder::default()
+            .with_line_protocol(&lp)
+            .with_max_seq(1);
```

Comment:
After you add the missing change in order function, we will need to add compaction level here to have it work correctly
```suggestion
            .with_max_seq(1)
            .with_compaction_level(CompactionLevel::Initial);
```

Reply:
[The default level set in the `TestParquetFileBuilder` is `CompactionLevel::Initial`.](https://github.com/influxdata/influxdb_iox/blob/50186ef5ee76f30b1b58a1890ce10a4ef9cd119d/iox_tests/src/util.rs#L620)